### PR TITLE
Fix Presto Limit

### PR DESCRIPTION
**Pull request by yanhan0502**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.
https://github.com/yahoo/fili/issues/1129
Fix nested cases. e.g. 
Before fix: 
`... FETCH NEXT 1 ROWS ONLY) AS "t4"`  ->  `LIMIT 1`
After fix:
`... FETCH NEXT 1 ROWS ONLY) AS "t4"`  ->  `LIMIT 1) AS "t4"`

### CODE REVIEW

**Comment by yanhan0502** [comment]
SD complains about test failures however, `mvn clean test` works fine in local machine.

**Comment by zmz0305** [comment]
Some tests in fili-core failed... Not sure why

**Comment by Ruiqian1** [comment]
Can we also add an entry to https://github.com/yahoo/fili/blob/master/CHANGELOG.md

**Comment by yanhan0502** [comment]
Looks like SD build is not stable. 
The first commit failed at `17:51:35 [INFO] Fili: Core Libraries ............................... FAILURE [01:43 min]`
But the second commit now failed at 
`09:22:02 [INFO] Fili: Luthier Config Libraries ..................... FAILURE [  2.637 s]`

**Review by Ruiqian1 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ Current
 -------
 ### Fixed:
 
+- [Fix: Presto Limit](https://github.com/yahoo/fili/pull/1128)
+
```
Comment (michael-mclawhorn) [inline CHANGELOG.md:13]: * Create a github issue and link to it for the feature.
* Make note of the tests that you're ignoring and removing and why (flaky on the first and non-compiling on the second)

File: fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/PrestoSqlBackedClient.java [hunk 2]
```diff
@@ -246,13 +246,7 @@ private static String fetchToLimitHelper(String fixQuotePrestoQuery) {
...
                 }
             }
-            int idxOfLineReturn = nextPart.indexOf('\n');
-            if (idxOfLineReturn != -1) {
-                nextPart = "LIMIT " + rows + nextPart.substring(idxOfLineReturn);
-
-            } else {
-                nextPart = "LIMIT " + rows;
-            }
+            nextPart = nextPart.replace("FETCH NEXT " + rows + " ROWS ONLY", "LIMIT " + rows);
```
Comment (michael-mclawhorn) [inline fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/PrestoSqlBackedClient.java:249]: Is there any way to provide unit test coverage for this change?

**Review by michael-mclawhorn — CHANGES_REQUESTED** [review]
On the documentation sides, some better CHANGELOG
On the testing side, I don't know if it's possible, but I always try to put in a confirmatory test on any fixes if at all possible.

File: CHANGELOG.md (hunk 1 above)
Comment (yanhan0502) [reply CHANGELOG.md:13]: Sure. Created https://github.com/yahoo/fili/issues/1129

File: fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/PrestoSqlBackedClient.java (hunk 2 above)
Comment (yanhan0502) [reply fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/PrestoSqlBackedClient.java:249]: Added a UT for this class and included *Test.java in maven-surefire-plugin to enable the test.

**Review by michael-mclawhorn — APPROVED** [review]
Looks good. 
It might be an interesting experience to try rewriting your test in spock to see how much more civilized multiline strings can be.  Regardless everything is good to go in terms of contribution guidelines, so here goes.

**Pull request merged**